### DISCUSSION TITLE
PrismaClientInitializationError when deploying Next14.2.2 and Prisma 5.14.0 using Azure SQL Database and Azure App Service

### DISCUSSION THREAD

**Discussion by akatsu-y2** [prisma/prisma#24389] [Q&A]
### Question

- I am deploying a Next14 application in a Linux environment on Azure App Service and trying to access an Azure SQL Database using Prisma 5.14.0. 
- However, I get a PrismaClientInitializationError and cannot access the database.
- Therefore, the logs were output by Azure App Service

- Result of SSH connection
 ```
/home/site/wwwroot/node_modules/prisma# ls
LICENSE    libquery_engine-debian-openssl-1.1.x.so.node       libquery_engine-linux-musl-arm64-openssl-3.0.x.so.node  package.json   query_engine-windows.dll.node
README.md  libquery_engine-debian-openssl-3.0.x.so.node       libquery_engine-linux-musl-openssl-3.0.x.so.node        preinstall     scripts
build      libquery_engine-linux-arm64-openssl-3.0.x.so.node  libquery_engine-linux-musl.so.node                      prisma-client
 ```
 
 ```
/home/site/wwwroot/node_modules/.prisma/client# ls
default.d.ts  edge.js           libquery_engine-debian-openssl-1.1.x.so.node            libquery_engine-linux-musl-openssl-3.0.x.so.node  schema.prisma
default.js    index-browser.js  libquery_engine-debian-openssl-3.0.x.so.node            libquery_engine-linux-musl.so.node                wasm.d.ts
deno          index.d.ts        libquery_engine-linux-arm64-openssl-3.0.x.so.node       package.json                                      wasm.js
edge.d.ts     index.js          libquery_engine-linux-musl-arm64-openssl-3.0.x.so.node  query_engine-windows.dll.node
 ```

[debug info.txt](https://github.com/user-attachments/files/15527675/debug.info.txt)

### How to reproduce (optional)

1. Create an Azure App Service in Linux environment with basic plan
2. Create a Free plan SQL Server in SQL Database
3. Build and deploy Next14 series apps with vscode, without using CI/CD such as github for that.
-  package.json
 ```
 "scripts": {
    "dev": "next dev",
    "build": "prisma generate && next build",
    "start": "node_modules/next/dist/bin/next start",
    "lint": "next lint",
    "prismaDBPush": "dotenv -e .env.local -- npx prisma db push",
    "prismaStudio": "dotenv -e .env.local -- npx prisma studio"
  },
```
4. Accessing the site raises a PrismaClientInitializationError

### Expected behavior (optional)

Prisma Client is successfully created

### Information about Prisma Schema, Client Queries and Environment (optional)

ex) schema.prisma
```prisma
generator client {
  provider = "prisma-client-js"
  binaryTargets = ["native", "debian-openssl-1.1.x",  "debian-openssl-3.0.x", "linux-musl", "linux-musl-openssl-3.0.x","linux-musl-arm64-openssl-3.0.x","linux-arm64-openssl-3.0.x"]
}

datasource db {
  provider = "sqlserver"
  url      = env("DATABASE_URL")
}

model Summary {
  id                String     @id @default(uuid())
  content           String
  status            Status     @relation(fields: [statusId], references: [id])
  statusId          String
  storageDocTitle   String
  documentTitle     String
  deleted           Boolean    @default(false)
  createdBy         User       @relation("SummaryCreatedBy", fields: [createdById], references: [id], onDelete: NoAction, onUpdate: NoAction)
  createdById       String
  createdAt         DateTime   @default(now())
  finishedAt        DateTime?
  updatedBy         User       @relation("SummaryUpdatedBy", fields: [updatedById], references: [id], onDelete: NoAction, onUpdate: NoAction)
  updatedById       String
  updatedAt         DateTime   @default(now())
  DocumentTag       DocumentTag[]
}
```

```ts
import { prisma } from '@/lib/prisma';
import { NextRequest, NextResponse } from 'next/server';

export const dynamic = 'force-dynamic';

export async function GET(req: NextRequest) {
  const searchParams = req.nextUrl.searchParams;
  const query = searchParams.get('query');

  try {
    let summaries;
    if (query) {
      summaries = await prisma.summary.findMany({
        where: {
          documentTitle: {
            contains: query,
        },
      });
    } else {
      summaries = await prisma.summary.findMany();
    }
    return NextResponse.json(summaries, { status: 200 });
  } catch (error) {
    console.error(error);
    return NextResponse.json({ message: error }, { status: 500 });
  }
}
```

- OS: Linux(Debian 12)
- Database: SQL Server(SQL Database)
- Node.js version: v20.11.1
- Next.js version: 14.2.2
- Prisma version: 5.14.0
- package manager : npm 10.2.4
- TypeScript
- Azure App Service
  - node 20 LTS
  - Start up command: npm run start

**Comment by ludralph** [comment]
HI @akatsu-y2 👋 

Thank you for raising this question.

According to the logs in the `debug info.txt` file you shared, this is the error you encountered
```
This is likely caused by tooling that has not copied "libquery_engine-debian-openssl-3.0.x.so.node" to the deployment folder.
Prisma Client could not locate the Query Engine for runtime "debian-openssl-3.0.x".
We detected that you are using Next.js, learn how to fix this: https://pris.ly/d/engine-not-found-nextjs.
```
Did you check out the recommendation on fixing this issue from this [documentation](https://www.prisma.io/docs/orm/more/help-and-troubleshooting/help-articles/nextjs-prisma-client-monorepo)?

**Reply by akatsu-y2** [reply]
Thank you for your response.

Of course, I tried using `@prisma/nextjs/nextjs-monorepo-workaround-plugin`, but no change.
Does this mean that when I use this, the client folder is automatically copied inside the prisma folder?
If so, I may be doing something wrong because the client folder was not generated when I used this plugin.

In the firewall settings of Azure SQL Database, if I check the “Allow Azure services and resources to access this server” as an exception to the firewall rule, I can access it.
However, if you do so, the IP address restriction will not work and the DB can be accessed from any IP.

**Reply by ludralph** [reply]
Hi @akatsu-y2 

The `@prisma/nextjs-monorepo-workaround-plugin` is designed to help with issues related to Prisma client generation and schema location in Next.js monorepo setups. The plugin is meant to copy necessary Prisma files (like schema.prisma and query engine) to the correct location during the Next.js build process, not to generate the client itself. You still need to run `prisma generate` to create the Prisma client before using the plugin.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
